### PR TITLE
Added support for multiple data types in length function

### PR DESCRIPTION
**Pull request by prajwalpatil2505**
Added support for nchar, nvarchar  and name data types in length function and corrected bytea and text types to return default length of -1. Modfied header size as for IPS_CLIENT it is 16.

### CODE REVIEW

File: rows.go [hunk 1]
```diff
@@ -49,9 +48,11 @@ func (fd fieldDesc) Name() string {
 func (fd fieldDesc) Length() (length int64, ok bool) {
 	switch fd.OID {
 	case oid.T_text, oid.T_bytea:
-		return math.MaxInt64, true
-	case oid.T_varchar, oid.T_bpchar:
+		return 4096, true
```
Comment (aniket-s-kulkarni) [inline rows.go:51]: @abhishekjog why 4096?

File: rows.go [hunk 2]
```diff
@@ -49,9 +48,11 @@ func (fd fieldDesc) Name() string {
...
+	case oid.T_varchar, oid.T_bpchar, oid.T_nvarchar, oid.T_nchar:
 		return int64(fd.Mod - headerSize), true
+	case oid.T_name:
+		return 256, true
```
Comment (aniket-s-kulkarni) [inline rows.go:55]: Ditto - why a fixed 256.

**Review by aniket-s-kulkarni — CHANGES_REQUESTED** [review]

File: rows.go (hunk 1 above)
Comment (abhishekjog) [reply rows.go:51]: TEXT is internal data type with variable length up to 65535. Length of this data type cannot be fetched. If length goes beyond 65535, table is toast. We can display it as -1 as seen in _t_type in catalog.

File: rows.go (hunk 2 above)
Comment (abhishekjog) [reply rows.go:55]: NAME is a internal datatype. and width is fixed at 256 chars.

**Review by abhishekjog — APPROVED** [review]

**Comment by prajwalpatil2505** [comment]
Table details:

```
                             Table "TABLETYP"
  Attribute   |              Type              | Modifier | Default Value
--------------+--------------------------------+----------+---------------
 BOOL1        | BOOLEAN                        |          |
 CHAR1        | CHARACTER(1)                   |          |
 INT81        | BIGINT                         |          |
 INT21        | SMALLINT                       |          |
 INT41        | INTEGER                        |          |
 FLOAT41      | REAL                           |          |
 FLOAT81      | DOUBLE PRECISION               |          |
 BPCHAR1      | CHARACTER(1)                   |          |
 VARCHAR1     | CHARACTER VARYING(10)          |          |
 DATE1        | DATE                           |          |
 TIME1        | TIME                           |          |
 TIMESTAMP1   | TIMESTAMP                      |          |
 TIMETZ1      | TIME WITH TIME ZONE            |          |
 NUMERIC1     | NUMERIC(18,0)                  |          |
 INT12        | BYTEINT                        |          |
 NCHAR1       | NATIONAL CHARACTER(1)          |          |
 NVARCHAR1    | NATIONAL CHARACTER VARYING(10) |          |
 ST_GEOMETRY1 | ST_GEOMETRY(10)                |          |
 VARBINARY1   | BINARY VARYING(10)             |          |
 JSON1        | JSON                           |          |
```

Output from nzgo:
```
ci=0 name=BOOL1 len=0:false dlen=0:0:false d=BOOL
ci=1 name=CHAR1 len=1:true dlen=0:0:false d=BPCHAR
ci=2 name=INT81 len=0:false dlen=0:0:false d=INT8
ci=3 name=INT21 len=0:false dlen=0:0:false d=INT2
ci=4 name=INT41 len=0:false dlen=0:0:false d=INT4
ci=5 name=FLOAT41 len=0:false dlen=0:0:false d=FLOAT4
ci=6 name=FLOAT81 len=0:false dlen=0:0:false d=FLOAT8
ci=7 name=BPCHAR1 len=1:true dlen=0:0:false d=BPCHAR
ci=8 name=VARCHAR1 len=10:true dlen=0:0:false d=VARCHAR
ci=9 name=DATE1 len=0:false dlen=0:0:false d=DATE
ci=10 name=TIME1 len=0:false dlen=0:0:false d=TIME
ci=11 name=TIMESTAMP1 len=0:false dlen=0:0:false d=TIMESTAMP
ci=12 name=TIMETZ1 len=0:false dlen=0:0:false d=TIMETZ
ci=13 name=NUMERIC1 len=0:false dlen=18:0:true d=NUMERIC
ci=14 name=INT12 len=0:false dlen=0:0:false d=INT1
ci=15 name=NCHAR1 len=1:true dlen=0:0:false d=NCHAR
ci=16 name=NVARCHAR1 len=10:true dlen=0:0:false d=NVARCHAR
ci=17 name=ST_GEOMETRY1 len=0:false dlen=0:0:false d=ST_GEOMETRY
ci=18 name=VARBINARY1 len=0:false dlen=0:0:false d=VARBINARY
ci=19 name=JSON1 len=0:false dlen=0:0:false d=JSON
```

**Comment by aniket-s-kulkarni** [comment]
Looks good @prajwalpatil2505

**Review by abhishekjog — APPROVED** [review]

**Pull request merged**